### ISSUE TITLE
Distributed downloads without HDFS

### ISSUE THREAD

**Issue by nilesh-c** [dbpedia/distributed-extraction-framework#55]
Say we're dealing with an N-machine cluster. We have M wikimedia mirrors. (currently M=4)

The initial plan was to assume HDFS has been setup over the N nodes (note that when I say HDFS, it also includes S3 - Hadoop makes using S3 in lieu of HDFS seamless without any code changes Re: https://wiki.apache.org/hadoop/AmazonS3)

* **Case A:** What **this issue** is about:
Support distributed downloads without involving a Hadoop/HDFS cluster. Suppose we have a guy who wants to leverage 10 machines with 10 IPs each (very unlikely scenario, mind) to download Wikipedia dumps in parallel *without* involving HDFS. It involves around 4-5 more hours of coding+testing but it's doable.
* **Case B:** In the more likely scenario where said guy has a single IP for all his machines (LAN in a lab), unless he wants to do distributed processing (which in turn needs HDFS etc.) he has no reason to use multiple machines. He can simply use a single node with M*2 workers. So for 4 mirrors, he runs 1 master and 8 workers on the same machine. Sounds easy. AND: we support this by default.

Question is, do we need to support **Case A**? If yes, please suggest a use-case for it (because the situation I described seems a bit far-fetched, maybe) ?

Sang and I were discussing this. @sangv @jimkont please share your views on this for both (a) right now, within GSoC, and if not now, (b) maybe later yes/no?

**Comment by nilesh-c** [comment]
So, for this year gsoc: we need to port B (parallel language downloading on single machine) to the main framework.

Also, I guess we reached the conclusion that we don't need HDFS-less distributed downloading for now? That is to say, downloading on multiple machines in parallel without having to install HDFS. Can't see any need for that. Therefore, closing issue.
